etna/libs/tstcc/tstcc.py

```diff
@@ -0,0 +1,276 @@
+            )
+        else:
+            data_loader = DataLoader(
+                dataset=dataset,
+                batch_size=self.batch_size,
+                shuffle=False,
+                drop_last=False,
+                num_workers=self.num_workers
+            )
+        return data_loader
+
+    def fit(self, train_data, n_epochs, lr, temperature, lambda1, lambda2):
+        """
+        Fit model
+
+        Parameters
+        ----------
+        train_data:
+            train data
+        n_epochs:
+            The number of epochs. When this reaches, the training stops.
+        lr:
+            The learning rate.
+        temperature:
+            Temperature in NTXentLoss.
+        lambda1:
+            The relative weight of the first item in the loss (temporal contrasting loss).
+        lambda2:
+            The relative weight of the second item in the loss (contextual contrasting loss).
+        """
+        train_loader = self.prepare_data(data=train_data, mode="train")
+        model_optimizer = torch.optim.Adam(self.model.parameters(), lr=lr, betas=(0.9, 0.99),
+                                           weight_decay=3e-4)
+        self.model.train()
+        for epoch in range(1, n_epochs + 1):
+
+            total_loss = []
+            for batch_idx, (aug1, aug2) in enumerate(train_loader):
+                # send to device
+                aug1, aug2 = aug1.to(self.device), aug2.to(self.device)
+
+                # optimizer
+                model_optimizer.zero_grad()
+
+                features1 = self.model.encoder(aug1)
+                features2 = self.model.encoder(aug2)
+
+                # normalize projection feature vectors
+                features1 = F.normalize(features1, dim=1)
+                features2 = F.normalize(features2, dim=1)
+
+                temp_cont_loss1, temp_cont_lstm_feat1 = self.model.temporal_contr_model(features1, features2)
+                temp_cont_loss2, temp_cont_lstm_feat2 = self.model.temporal_contr_model(features2, features1)
+
+                # normalize projection feature vectors
+                zis = temp_cont_lstm_feat1
+                zjs = temp_cont_lstm_feat2
+
+                # compute loss
+                nt_xent_criterion = NTXentLoss(
+                    device=self.device,
+                    batch_size=self.batch_size,
+                    temperature=temperature,
+                    use_cosine_similarity=self.use_cosine_similarity
+                )
+                loss = (temp_cont_loss1 + temp_cont_loss2) * lambda1 + nt_xent_criterion(zis, zjs) * lambda2
+
+                total_loss.append(loss.item())
+                loss.backward()
+                model_optimizer.step()
+
+            train_loss = torch.tensor(total_loss).mean()
+
+            print(f'\nEpoch : {epoch}\n'
+                  f'Train Loss     : {train_loss:.4f}\n')
+
+        print("\n################## Training is Done! #########################")
```

Comment:
Are we ok with this printing?

Reply:
<img width="513" alt="image" src="https://github.com/etna-team/etna/assets/82458209/42c90cc9-c8fe-40af-a2a0-6f2e2148dc4a">

Reply:
I'm not sure there should be any prints. It just clutters the output. Or we should do it using logger.

It also creates inconsistency between our transforms.

Reply:
When I do training in notebook, cell output is repeated like this after each following `fit` calling (first calling doesn't have such problem)
<img width="796" alt="Screenshot 2024-04-08 at 12 59 09" src="https://github.com/etna-team/etna/assets/82458209/64c9621a-c780-4426-80ce-855be072f90e">